lib/checkPullRequestJob.js

```diff
@@ -0,0 +1,210 @@
+      'endeavour to add the new job ' + fileText + ' to the job registry and ';
+  }
+
+  return registryReminder;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} newJobFiles
+ */
+const getJobNameString = (newJobFiles) => {
+  let jobNameString = '';
+  const jobNameArray = newJobFiles.map((file) =>
+    getJobNameFromFileName(file.filename)
+  );
+  if (jobNameArray.length === 1) {
+    jobNameString = ' The name of the job is ' + jobNameArray[0] + '.';
+  } else {
+    jobNameString = ' The jobs are ' + jobNameArray.join(', ') + '.';
+  }
+
+  return jobNameString;
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} newJobFiles
+ * @param {import('probot').Octokit.PullsListFilesResponseItem[]} changedFiles
+ * @param {string} prAuthor - Author of the Pull Request
+ */
+const getCommentBody = (newJobFiles, changedFiles, prAuthor) => {
+  const registryReminder = getRegistryReminder(newJobFiles, changedFiles);
+  const jobNameString = getJobNameString(newJobFiles);
+  const newLineFeed = '<br>';
+  const serverJobsForm = 'server jobs form'.link(
+    'https://goo.gl/forms/XIj00RJ2h5L55XzU2');
+
+  // This function will never be called when there are no job files.
+  if (newJobFiles.length === 1) {
+    const serverAdminPing = 'Hi @' + SERVER_JOBS_ADMIN + ', PTAL at this PR, ' +
+      'it adds a new server job.' + jobNameString;
+
+    const prAuthorPing = 'Also @' + prAuthor + registryReminder +
+    'please make sure to fill in the ' + serverJobsForm +
+    ' for the new job to be tested on the backup server. ' +
+    'This PR can be merged only after the test is successful.';
+
+    return (serverAdminPing + newLineFeed + prAuthorPing +
+      newLineFeed + 'Thanks!');
+  } else {
+    const serverAdminPing = 'Hi @' + SERVER_JOBS_ADMIN + ', PTAL at this PR, ' +
+      'it adds new server jobs.' + jobNameString;
+
+    const prAuthorPing = 'Also @' + prAuthor + registryReminder +
+    'please make sure to fill in the ' + serverJobsForm +
+    ' for the new jobs  to be tested on the backup server.' +
+    ' This PR can be merged only after the test is successful.';
+
+    return (serverAdminPing + newLineFeed + prAuthorPing +
+      newLineFeed + 'Thanks!');
+  }
+};
+
+/**
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasCriticalLabel = (pullRequest) => {
+  const criticalLabel = pullRequest.labels.find(
+    (label) => label.name.toLowerCase() === 'critical'
+  );
+
+  return Boolean(criticalLabel);
+};
+
+/**
+ * @param {import('probot').Context} context
+ */
+const checkForNewJob = async (context) => {
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse} pullRequest
+   */
+  const pullRequest = context.payload.pull_request;
+  if (!hasCriticalLabel(pullRequest)) {
+    const changedFiles = await getAllChangedFiles(context);
+
+    // Get new jobs that were created in the PR.
+    const newJobFiles = changedFiles.filter((file) => {
```

Comment:
We can have one more case, a new job being added in an existing job file, right?

Reply:
Yes.